src/main/files.ts

```diff
@@ -51,31 +50,23 @@ export async function showSaveDialog(event?: IpcEvents, as?: string) {
   console.log(`Asked to save to ${filePaths[0]}`);
 
   // Let's confirm real quick if we want this
-  if (await ensureSaveTargetEmpty(filePaths[0])) {
+  if (await isOkToSaveAt(filePaths[0])) {
     ipcMainManager.send(event || IpcEvents.FS_SAVE_FIDDLE, [filePaths[0]]);
   }
 }
 
 /**
- * Ensures that a folder designated for saving is empty
+ * Confirm it's OK to save files in `folder`
  *
  * @param {string} filePath
  * @returns {Promise<boolean>}
  */
-async function ensureSaveTargetEmpty(filePath: string): Promise<boolean> {
-  const targetPaths = Object.values(DefaultEditorId).map((filename) =>
-    path.join(filePath, filename),
+async function isOkToSaveAt(filePath: string): Promise<boolean> {
+  return (
+    !fs.existsSync(filePath) ||
+    fs.readdirSync(filePath).filter(isSupportedFile).length === 0 ||
```

Comment:
nit(non-blocking): If the function is `async`, is there value to using the synchronous `fs` APIs here? Is the performance cost negligible either way?

Reply:
I don't think there's any harm in using async here. Updated in 41260f3fa427794ab13c8fc1455e68f3f32e8932 .